tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetRequest.scala

```diff
@@ -0,0 +1,112 @@
+
+case class UnparsedPublicAssetId(id: String)
+
+object PublicAssetCreationResponse {
+  def from(publicAsset: PublicAsset): PublicAssetCreationResponse =
+    PublicAssetCreationResponse(publicAsset.id,
+      publicAsset.blobId,
+      publicAsset.publicURI,
+      publicAsset.size.value,
+      publicAsset.contentType,
+      publicAsset.identityIds)
+}
+
+case class PublicAssetCreationResponse(id: PublicAssetId,
+                                       blobId: BlobId,
+                                       publicURI: PublicURI,
+                                       size: Long,
+                                       contentType: ImageContentType,
+                                       identityIds: Seq[IdentityId])
+
+case class PublicAssetSetResponse(accountId: AccountId,
+                                  oldState: Option[UuidState],
+                                  newState: UuidState,
+                                  created: Option[Map[PublicAssetCreationId, PublicAssetCreationResponse]],
+                                  notCreated: Option[Map[PublicAssetCreationId, SetError]])
+
+case class PublicAssetCreationParseException(setError: SetError) extends PublicAssetException {
+  override val message: String = s"Invalid public asset creation request: ${setError.description}"
+}
+
+case class PublicAssetInvalidBlobIdException(blobId: String) extends PublicAssetException {
+  override val message: String = s"Invalid blobId: $blobId"
+}
+
+case class PublicAssetBlobIdNotFoundException(blobId: String) extends PublicAssetException {
+  override val message: String = s"BlobId not found: ${blobId}"
+}
+
+case class PublicAssetInvalidIdentityIdException(identityId: String) extends PublicAssetException {
+  override val message: String = s"Invalid identityId: $identityId"
+}
+
+case class PublicAssetIdentityIdNotFoundException(identityId: String) extends PublicAssetException {
+  override val message: String = s"IdentityId not found: ${identityId}"
+}
+
+sealed trait PublicAssetCreationResult {
+  def publicAssetCreationId: PublicAssetCreationId
+}
+
+case class PublicAssetCreationSuccess(publicAssetCreationId: PublicAssetCreationId, publicAssetCreationResponse: PublicAssetCreationResponse) extends PublicAssetCreationResult
+
+case class PublicAssetCreationFailure(publicAssetCreationId: PublicAssetCreationId, exception: Throwable) extends PublicAssetCreationResult {
+  def asPublicAssetSetError: SetError = exception match {
+    case e: PublicAssetCreationParseException => e.setError
+    case e: PublicAssetException if (e.isInstanceOf[PublicAssetInvalidContentTypeException]
+      || e.isInstanceOf[PublicAssetInvalidBlobIdException]
+      || e.isInstanceOf[PublicAssetBlobIdNotFoundException]
+      || e.isInstanceOf[PublicAssetInvalidIdentityIdException]
+      || e.isInstanceOf[PublicAssetIdentityIdNotFoundException]) => SetError.invalidArguments(SetError.SetErrorDescription(e.getMessage))
```

Comment:
This big if looks unecessary conplicated to me.

Make them all subclass of PublicAssetArgumentException and use only PublicAssetArgumentException in the pattern matching?